postproengine/spod.py

```diff
@@ -0,0 +1,685 @@
+        t = np.asarray(np.array(db['times']).data)
+        for iplane in iplanes:
+            xc[iplane] = np.asarray(db['x'].data[iplane])
+            udata[iplane] = np.zeros((len(t),len(z),len(y),3))
+            udata[iplane][:,:,:,0] = np.array(db['velocityx'][:,iplane,:,:])
+            udata[iplane][:,:,:,1] = np.array(db['velocityy'][:,iplane,:,:])
+            udata[iplane][:,:,:,2] = np.array(db['velocityz'][:,iplane,:,:])
+    else:
+        find_nearest = lambda a, a0: np.abs(np.array(a) - a0).argmin()
+        iters = [find_nearest(timevec, t) for t in trange]
+        iters = np.arange(iters[0],iters[1]+1)
+        db = ppsamplexr.getPlaneXR(ncfile,iters,varnames,groupname=group,verbose=0,includeattr=True,gettimes=True)
+        YY = np.array(db['y'])
+        ZZ = np.array(db['z'])
+
+        flow_index  = -np.ones(3)
+        for i in range(0,3):
+            if (sum(db['axis'+str(i+1)]) != 0):
+                flow_index[i] = np.nonzero(db['axis'+str(i+1)])[0][0]
+        for i in range(0,3):
+            if flow_index[i] == -1:
+                flow_index[i] = 3 - (flow_index[i-1] + flow_index[i-2])
+
+        streamwise_index = 2-np.where(flow_index == 0)[0][0]
+        lateral_index = 2-np.where(flow_index == 1)[0][0]
+        vertical_index = 2-np.where(flow_index == 2)[0][0]
+
+        slices = [slice(None)] * 3
+        slices[streamwise_index] = 0
+        slices[vertical_index] = 0
+        y = YY[tuple(slices)]
+
+        slices = [slice(None)] * 3
+        slices[streamwise_index] = 0
+        slices[lateral_index] = 0
+        z = ZZ[tuple(slices)]
+
+        permutation = [streamwise_index, vertical_index, lateral_index]
+
+        t = np.asarray(np.array(db['times']).data)
+        for iplane in iplanes:
+            xc[iplane] = np.asarray(db['x'][iplane][0,0])
+            udata[iplane] = np.zeros((len(t),len(z),len(y),3))
+            for i,tstep in enumerate(iters):
+                ordered_data = np.transpose(np.array(db['velocityx'][tstep]),permutation)
+                udata[iplane][i,:,:,0] = ordered_data[iplane,:,:]
+
+                ordered_data = np.transpose(np.array(db['velocityy'][tstep]),permutation)
+                udata[iplane][i,:,:,1] = ordered_data[iplane,:,:]
+
+                ordered_data = np.transpose(np.array(db['velocityz'][tstep]),permutation)
+                udata[iplane][i,:,:,2] = ordered_data[iplane,:,:]
+
+    return udata , y , z , t 
+
+def interpolate_radial_to_cart(U,rr,theta,YY,ZZ,offsety,offsetz):
+    YR = np.sqrt(np.square((YY-offsety)) + np.square((ZZ-offsetz)))
+    ZR = np.arctan2((ZZ-offsetz),(YY-offsety))
+    ZR[ZR < 0] += 2 * np.pi
+    grid = (YR,ZR)
+    positions = np.vstack(list(map(np.ravel,grid))).T
+    U_interp = sp.interpolate.interpn((rr,theta),U,positions,method='linear',bounds_error=False,fill_value = 0)
+    U_interp = np.reshape(U_interp,(YR.shape[0],ZR.shape[1]))
+    return U_interp
+
+def interpolate_cart_to_radial(U,yy,zz,RR,TT,offsety,offsetz):
```

Comment:
yes, `interpolate_cart_to_radial()` is a good one to move to `__init__.py` so that it can be used by other executors.  I was thinking to use `udfmodules` when there is a particular geometry or user-specific function that is only applicable to a specific domain.  FFT's and coordinate system changes might be general enough to be reused as a part of the postprocessing engine.